File: Week-01/02-Backspace String Compare/Ashraf.cpp
// Author: Ashraf
// Link: https://leetcode.com/problems/backspace-string-compare/
// Time Complexity: O(n)
// Space Complexity: O(n)
class Solution {
public:
    bool backspaceCompare(string s, string t) {
     string temp1 = "", temp2 = "";   
     for (auto it : s) {
            if (it == '#') {
                if (!temp1.empty()) {
                    temp1.pop_back();
                }
            } else {
                temp1 += it;
            }
        }
        for (auto it : t) {
            if (it == '#') {
                if (!temp2.empty()) {
                    temp2.pop_back();
                }
            } else {
                temp2 += it;
            }
        }
        return temp1 == temp2;
    }
};

File: Week-01/01-Baseball Game/Ashraf.java
// Author: Ashraf
// Link: https://leetcode.com/problems/baseball-game/
// Time Complexity: O(n)
// Space Complexity: O(n)
class Solution {
    public int calPoints(String[] operations) {
        List<Integer> temp = new ArrayList<>();
        for (String it : operations) {
            if (it.equals("+")) {
                temp.add(temp.get(temp.size() - 1) + temp.get(temp.size() - 2));
            } else if (it.equals("D")) {
                temp.add(temp.get(temp.size() - 1) * 2);
            } else if (it.equals("C")) {
                temp.remove(temp.size() - 1);
            } else {
                temp.add(Integer.parseInt(it));
            }
        }
        int sum = 0;
        for (int it : temp) {
            sum += it;
        }
        return sum;
    }
}